proof/refine/Invariants_H.thy

```diff
@@ -374,12 +312,18 @@ where valid_cap'_def:
   | ThreadCap r \<Rightarrow> tcb_at' r s
   | ReplyCap r m x \<Rightarrow> tcb_at' r s
   | IRQControlCap \<Rightarrow> True
-  | IRQHandlerCap irq \<Rightarrow> irq \<le> maxIRQ \<and> irq \<noteq> irqInvalid
+  | IRQHandlerCap irq \<Rightarrow> arch_valid_irq irq \<comment> \<open>arch-dependent maxIRQ bound and invalidIRQ constraint\<close>
```

Comment:
This looks like the `maxIRQ` condition is now in arch_valid_irq as well -- that one is generic, we'll want to use it for generic invariant proofs where the value of `maxIRQ` doesn't matter (the value is arch specific, the concept is generic). The `irqInvalid` reasoning is only needed once, but `maxIRQ` happens more often.

Reply:
Oh. That is a good point. Do you want me to change it now, or add a FIXME?

Reply:
I'm fine to merge it now if we fix it up as one of the next steps -- if we leave it much later it's likely to become ingrained.

Reply:
(Unblocking the PR queue would be good, though, so maybe separate PR is good)

Reply:
Ok. I agree. If you want you can also do it in a next PR, it's going to be pretty easy: move the maxIRQ back, update the arch-specific definitions (to \<top> for non-RISCV64), and then I think at this stage of arch-split the proofs should go through with minimal or no fixes.